Firmware/RTK_Surveyor/Tasks.ino

```diff
@@ -350,28 +387,43 @@ void handleGNSSDataTask(void *e)
 
           if (settings.enablePrintSDBuffers && !inMainMenu)
           {
-            int availableUARTSpace = settings.uartReceiveBufferSize - serialGNSS.available();
-            systemPrintf("SD Incoming Serial: %04d\tToRead: %04d\tMovedToBuffer: %04d\tavailableUARTSpace: %04d\tavailableHandlerSpace: %04d\tToRecord: %04d\tRecorded: %04d\tBO: %d\r\n", serialGNSS.available(), 0, 0, availableUARTSpace, availableHandlerSpace, sliceToRecord, 0, bufferOverruns);
+            int bufferAvailable;
+            if (USE_I2C_GNSS)
+              bufferAvailable = serialGNSS.available();
+            else
+            {
+              theGNSS.checkUblox();
+              bufferAvailable = theGNSS.fileBufferAvailable();
+            }
+            int availableUARTSpace;
+            if (USE_I2C_GNSS)
+              availableUARTSpace = settings.uartReceiveBufferSize - bufferAvailable;
+            else
+              // Use gnssHandlerBufferSize for now. TODO: work out if the SPI GNSS needs its own buffer size setting
+              availableUARTSpace = settings.gnssHandlerBufferSize - bufferAvailable;
+            systemPrintf("SD Incoming Serial: %04d\tToRead: %04d\tMovedToBuffer: %04d\tavailableUARTSpace: %04d\tavailableHandlerSpace: %04d\tToRecord: %04d\tRecorded: %04d\tBO: %d\r\n", bufferAvailable, 0, 0, availableUARTSpace, availableHandlerSpace, sliceToRecord, 0, bufferOverruns);
           }
 
           //Write the data to the file
           long startTime = millis();
 
           sdBytesToRecord = ubxFile->write(&ringBuffer[sdTail], sliceToRecord);
+          if (USE_MMC_MICROSD)
+            ubxFile->flush();
```

Comment:
Does flush = sync on MMC? We avoid flush/sync in SPI mode to avoid long writes and risk of dropping data. I presume MMC doesn't have the same problem. However, if you are flushing/sync here, then the lower code where we turn on the LED during a sync (to indicate card writes) is not really used. Are you seeing the LED blink at all? I suspect it's so fast as to be not noticeable. You may want to create a new (millis() - lastUBXLogSyncTime > 1000) or whatever for MMC syncing. I'm interested to hear any timing measurements you may do involving MMC syncing/flushing. SPI would vary widely depending on card and block within the same card.

Reply:
So, initially, I couldn't get the SD-logging icon to appear using SD_MMC. I only got the vertical bar. Turns out that SD_MMC only reports the correct (incremented) file size if you do a flush first. BUT this might only be an issue in 2.0.2. I bet they've (probably) fixed it in ~2.0.6.

I really like the idea of a periodic flush. So long as it keeps the SD-logging icon happy.